library/core/src/tuple.rs

```diff
@@ -129,13 +128,9 @@ const fn ordering_is_some(c: Option<Ordering>, x: Ordering) -> bool {
     // This isn't using `match` because that optimizes worse due to
```

Comment:
Yeah, this essentially uses
```rust
const fn ordering_is_some1(c: Option<Ordering>, x: Ordering) -> bool {
    x as i8 == match c {
        Some(c) => c as i8,
        None => 2,
    }
}
```
instead of
```rust
const fn ordering_is_some2(c: Option<Ordering>, x: Ordering) -> bool {
    match c {
        Some(c) => c as i8 == x as i8,
        None => false,
    }
}
```
, because the first one [optimizes better](https://rust.godbolt.org/z/8EPWKozxa).

Do you have a suggestion for a better comment?